compiler/rustc_builtin_macros/src/deriving/generic/mod.rs

```diff
@@ -680,6 +680,12 @@ impl<'a> TraitDef<'a> {
                     param_clone
                 }
             })
+            .map(|mut param| {
+                // Remove all attributes, because there might be helper attributes
+                // from other macros that will not be valid in the expanded implementation.
+                param.attrs = ThinVec::new();
```

Comment:
Right now, we clear the attrs of the param after having cloned the entire param. As a result we clone the attrs only to throw them away. However, avoiding this makes the code more awkward. I've just noticed that we also needlessly clone const parameter defaults. Er, this could be addressed in a separate PR, if need be.

Reply:
I think the compiler would optimize it away, and writing it like this is easier to read?

Either way, if this whole thing might be significantly changed in the near future by addressing the real bug here (#61552), I don't think it's necessary to do anything about it in this PR